elastic/eui
# [Emotion] Convert `EuiResizableCollapseButton`

## Summary

closes #6408 (final component)

Ohh boy, this is a fun one. There was a lot of very branch-y Sass to untangle that I ended up basically rewriting in https://github.com/elastic/eui/pull/7091/commits/ab23346cce9753bc2873ff6758a1b9f4e2ec3609, so buckle your seatbelts and as always, [follow along by commit](https://github.com/elastic/eui/pull/7091/commits)!

## QA

- `yarn compile-scss && yarn storybook`
- Go to http://localhost:6006/?path=/story/euiresizablecollapsebutton--production-usage
- Confirm that the `direction` and `position` behave as follows:
    - When the `direction` is `horizontal`:
        - [x] `top` displays the collapse button at the top when both collapsed and not collapsed
        - [x] `bottom` displays the collapse button at the bottom when both collapsed and not collapsed
        - [x] `middle`, `left`, and `right` display the collapse button vertically centered for all positions
    - When the `direction` is `vertical`:
        - [x] `left` displays the collapse button on left when both collapsed and not collapsed
        - [x] `right` displays the collapse button at the right when both collapsed and not collapsed
        - [x] `middle`, `top`, and `bottom` display the collapse button horizontally centered for all positions
        - [x] Change the position back to `left` and change the writing mode in the toolbar from `LTR` to `RTL`. Confirm that the button correctly displays on the right instead of the left
- [Optional] Go to http://localhost:6006/?path=/story/euiresizablecollapsebutton--playground and play with all props controls to get a sense of the component

### General checklist

- [x] Props have proper **autodocs** (using `@default` if default values are missing) and **storybook toggles**


<details><summary><h3>Emotion checklist</h3></summary>

**Kibana usage**

- [x] Search Kibana's codebase for `euiResizableToggleButton-` (case sensitive) to check for usage of modifier classes
  - ✅ **No usages whatsoever exist**

---

**General**


---

**Unit tests**

No tests added/written as this component is internal-only and did not previously have any tests

- ~[ ] Removed any `mount()`ed snapshots in favor of `render()` or a more specific assertion~

---

**Sass/Emotion conversion process**

- ~[ ] Removed component from `src/components/index.scss`~ - Not doable yet

---

**CSS tech debt**


---

**DOM Cleanup**


---

**Kibana due diligence**

- [x] Any test/query selectors that will need to be updated - **None**
- [x] Any Sass or CSS that will need to be updated, particularly if a component Sass var was deleted - **None**
- [x] Any direct className usages that will need to be refactored (e.g. someone calling the `euiBadge` class on a div instead of simply using the `EuiBadge` component) - **No usages**

---

**Extras/nice-to-have**

- ~[ ] Check for issues in the backlog that could be a quick fix for that component~ - Will be opening up a follow-up PR for this
</details>
